cmd/zoekt-sourcegraph-indexserver/sg.go

```diff
@@ -129,6 +140,94 @@ type sourcegraphClient struct {
 	configFingerprintReset time.Time
 }
 
+// GetRepoRank asks Sourcegraph for the score vector of repoName.
+func (s *sourcegraphClient) GetRepoRank(ctx context.Context, repoName string) ([]float64, error) {
+	u := s.Root.ResolveReference(&url.URL{
+		Path: "/.internal/ranks/" + strings.Trim(repoName, "/"),
+	})
+
+	b, err := s.get(ctx, u)
+	if err != nil {
+		return nil, err
+	}
+
+	var ranks []float64
+	err = json.Unmarshal(b, &ranks)
+	if err != nil {
+		return nil, err
+	}
+
+	return ranks, nil
+}
+
+// GetDocumentRanks asks Sourcegraph for a mapping of file paths to score
+// vectors.
+func (s *sourcegraphClient) GetDocumentRanks(ctx context.Context, repoName string) (map[string][]float64, error) {
+	u := s.Root.ResolveReference(&url.URL{
+		Path: "/.internal/ranks/" + strings.Trim(repoName, "/") + "/documents",
+	})
+
+	b, err := s.get(ctx, u)
+	if err != nil {
+		return nil, err
+	}
+
+	ranks := make(map[string][]float64)
+	err = json.Unmarshal(b, &ranks)
+	if err != nil {
+		return nil, err
+	}
+
+	// Invariant: All score vectors have the same length.
```

Comment:
I see you keep this invariant here. I'd still recommend somehow protecting this invariant in the code that can panic given this bit of code "lives" far away from the builder package.